paparazzi/paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PaparazziPlugin.kt

```diff
@@ -179,9 +179,11 @@ class PaparazziPlugin : Plugin<Project> {
         test.inputs.files(nativePlatformFileCollection)
           .withPropertyName("paparazzi.nativePlatform")
           .withPathSensitivity(PathSensitivity.NONE)
+        test.inputs.files(snapshotOutputDir)
+          .withPropertyName("paparazzi.snapshot.output.dir")
+          .withPathSensitivity(PathSensitivity.RELATIVE)
```

Comment:
```suggestion
        if (isVerifyRun.getOrElse(false)) {
          test.inputs.dir(snapshotOutputDir)
            .withPropertyName("paparazzi.snapshot.output.dir")
            .withPathSensitivity(PathSensitivity.RELATIVE)
        }
        if (isRecordRun.getOrElse(false)) {
          test.outputs.dir(snapshotOutputDir)
        }
```

Similar to @TWiStErRob, I would suggest wrapping this in ifs, as the task can run in different modes, which causes it to actually have different inputs & outputs. In verify mode, `snapshotOutputDir` is an input, as the captured snapshots are compared to the ones stored there. In record mode `snapshotOutputDir` is an output, as that's where the captured snapshots are copied. If in neither (running just `testDebug` for example), the `snapshotOutputDir` doesn't have anything to do with it, so it also shouldn't be tracked. When done like suggested, the only case where Gradle will still be confused is if one tries to run both `verifyPaparazzi` and `recordPaparazzi` tasks <- this should be solved by the copying from `build/reports/paparazzi` to `src/snapshots` to actually happen in the `recordPaparazzi` task and not in the test task, however that is bigger change. For now, I'd either crash the build if both are requested, or perhaps print a warning about it - the build will still go through, just caching will be disabled in this case.